EIPS/eip-5988.md

```diff
@@ -0,0 +1,159 @@
+---
+eip: 5988
+title: Add Poseidon hash function precompile
+description: Add a precompiled contract which implements the hash function used in the Poseidon cryptographic hashing algorithm
+author: Abdelhamid Bakhta (@abdelhamidbakhta), Eli Ben Sasson (@Elistark), Avihu Levy (@avihu28), David Levit Gurevich (@DavidLevitGurevich)
+discussions-to: https://ethereum-magicians.org/t/eip-5988-add-poseidon-hash-function-precompile/11772
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-11-15
+---
+
+## Abstract
+
+This EIP introduces a new precompiled contract which implements the hash function used in the Poseidon cryptographic hashing algorithm, for the purpose of allowing interoperability between the EVM and ZK / Validity rollups, as well as introducing more flexible cryptographic hash primitives to the EVM.
+
+## Motivation
+
+[Poseidon](../assets/eip-5988/papers/poseidon_paper.pdf) is an arithmetic hash function that is designed to be efficient for Zero-Knowledge Proof Systems.
+Ethereum adopts a rollup centric roadmap and hence must adopt facilities for L2s to be able to communicate with the EVM in an optimal manner.
+
+ZK-Rollups have particular needs for cryptographic hash functions that can allow for efficient verification of proofs.
+
+The Poseidon hash function is a set of permutations over a prime field, which makes it particularly well-suited for the purpose of building efficient ZK / Validity rollups on Ethereum.
+
+Poseidon is one of the most efficient hashing algorithms that can be used in this context.
+Moreover it is compatible with all major proof systems (SNARKs, STARKs, Bulletproofs, etc...).
+This makes it a good candidate for a precompile that can be used by many different ZK-Rollups.
+
+An important point to note is that ZK rollups using Poseidon have chosen different sets of parameters, which makes it harder to build a single precompile for all of them.
+
+However, we can still build a generic precompile that supports arbitrary parameters, and allow the ZK rollups to choose the parameters they want to use.
+
+This is the approach that we have taken in this EIP.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
```

Comment:
Generally, a table is added here with all the parameters, such as `POSEIDON_PRECOMPILE_ADDRESS` and `FORK_BLKNUM`. Also, `FORK_BLOCK_NUMBER` is often `FORK_BLKNUM`, although this is not required.

Reply:
done